src/main/java/de/tum/in/www1/artemis/domain/Result.java

```diff
@@ -257,7 +257,7 @@ public void setRated(Boolean rated) {
     }
 
     public void setRatedIfNotExceeded(@Nullable ZonedDateTime exerciseDueDate, ZonedDateTime submissionDate) {
-        this.rated = exerciseDueDate == null || submissionDate.isBefore(exerciseDueDate);
+        this.rated = exerciseDueDate == null || !submissionDate.isAfter(exerciseDueDate);
```

Comment:
In general it is recommended by clean code guides that you should avoid negation statements if possible as they make the code harder to understand: https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29#understandability-tips. Maybe you could just add a new OR condition for the equal case :)?